### DISCUSSION TITLE
How to build lancedb Python package?

### DISCUSSION THREAD

**Discussion by ChenZhongPu** [lancedb/lancedb#1327] [Q&A] [answered]
I followed the [guides](https://github.com/lancedb/lancedb/tree/main/python) to build the Python package, but it fails.

After running `python -m pip install .[tests,dev]`, the `lancedb` package was built and installed. But I failed to import `lancedb` in Python Shell:

> ImportError: cannot import name 'LanceDataset' from 'lance' (/home/xxx/venv/lib/python3.12/site-packages/lance/__init__.py)

BTW, will the command above also run `maturin develop` indirectly?

Neither `pytest` nor `pytest --doctest-modules python/lancedb` works. 

> E   importlib.metadata.PackageNotFoundError: No package metadata was found for lancedb

**Comment by wjones127** [comment, marked as answer]
Keep in mind, the `lance` and `pylance` packages are different. You want `pylance`, not `lance`. You might need to run `pip uninstall lance` and `pip install pylance` to get the correct package.

**Reply by ChenZhongPu** [reply]
Thanks for your hint :)
